### PR TITLE
Rely on resource group removal when performing destroy

### LINKED ISSUE #13: destroy can fail with InUseSubnetCannotBeDeleted
```
        TASK [Destroy subnet] **********************************************************
        task path: /private/var/folders/3q/pc1jcyjj3qqbj6_x7n4kbtqw0000gn/T/pytest-of-ssbarnea/pytest-30/test_command_init_scenario0/KKRMW/test-init/molecule/test-scenario/destroy.yml:32
        Monday 28 October 2019  17:13:23 +0000 (0:00:05.585)       0:00:06.313 ********
        <127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ssbarnea
        <127.0.0.1> EXEC /bin/sh -c 'echo ~ssbarnea && sleep 0'
        <127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608 `" && echo ansible-tmp-1572282803.54769-70190396942608="` echo /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608 `" ) && sleep 0'
        Using module file /Users/ssbarnea/os/ansible/lib/ansible/modules/cloud/azure/azure_rm_subnet.py
        <127.0.0.1> PUT /Users/ssbarnea/.ansible/tmp/ansible-local-130110rzk9pw7/tmp3rkmgqzv TO /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608/AnsiballZ_azure_rm_subnet.py
        <127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608/ /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608/AnsiballZ_azure_rm_subnet.py && sleep 0'
        <127.0.0.1> EXEC /bin/sh -c '/Users/ssbarnea/.pyenv/versions/3.7.4/bin/python3.7 /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608/AnsiballZ_azure_rm_subnet.py && sleep 0'
        <127.0.0.1> EXEC /bin/sh -c 'rm -f -r /Users/ssbarnea/.ansible/tmp/ansible-tmp-1572282803.54769-70190396942608/ > /dev/null 2>&1 && sleep 0'
        The full traceback is:
          File "/var/folders/3q/pc1jcyjj3qqbj6_x7n4kbtqw0000gn/T/ansible_azure_rm_subnet_payload__fxk56aq/ansible_azure_rm_subnet_payload.zip/ansible/modules/cloud/azure/azure_rm_subnet.py", line 372, in delete_subnet
          File "/Users/ssbarnea/.pyenv/versions/3.7.4/lib/python3.7/site-packages/azure/mgmt/network/v2018_08_01/operations/subnets_operations.py", line 108, in delete
            **operation_config
          File "/Users/ssbarnea/.pyenv/versions/3.7.4/lib/python3.7/site-packages/azure/mgmt/network/v2018_08_01/operations/subnets_operations.py", line 75, in _delete_initial
            raise exp

        fatal: [localhost]: FAILED! => {
            "changed": false,
            "invocation": {
                "module_args": {
                    "ad_user": null,
                    "address_prefix_cidr": null,
                    "adfs_authority_url": null,
                    "api_profile": "latest",
                    "auth_source": null,
                    "cert_validation_mode": null,
                    "client_id": null,
                    "cloud_environment": "AzureCloud",
                    "name": "molecule_subnet",
                    "password": null,
                    "profile": null,
                    "resource_group": "molecule",
                    "route_table": null,
                    "secret": null,
                    "security_group": null,
                    "service_endpoints": null,
                    "state": "absent",
                    "subscription_id": null,
                    "tenant": null,
                    "virtual_network_name": "molecule_vnet"
                }
            }
        }

        MSG:

        Error deleting subnet molecule_subnet - Azure Error: InUseSubnetCannotBeDeleted
        Message: Subnet molecule_subnet is in use by /subscriptions/d13dc0b1-1d61-4c2b-a416-bb5936f4b760/resourceGroups/molecule/providers/Microsoft.Network/networkInterfaces/instance01/ipConfigurations/default and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet.
```

### PR DESCRIPTION
**Pull request by mksha**
* Renamed azure resource group name to molecule-azure from molecule
* Removed instance and other resource destruction because we already have resource-group destruction tasks that will take care of the destruction of all resources within rg.

Fixes: #13

### CODE REVIEW

**Comment by mksha** [comment]
@ssbarnea can you approve it ?

**Review by ssbarnea — CHANGES_REQUESTED** [review]
- Please do not rename resource group, is the same everywhere.
- You removed the log line without any reason
- In general each change should be atomic and not combine multiple changes 
- Linking to a bug and using `Fix: #XYZ` as a comment is recommended. Still for this
  change alone (Rely on resource group removal on destroy) we may do it w/o that.

File: molecule_azure/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/destroy.yml [hunk 1]
```diff
@@ -10,50 +10,12 @@
...
-      # avoid failure when resource_group is already missing, as we remove it later anyway
-      failed_when: false
-
-    - name: Destroy resource group
+    - name: Destroy resource group and all associated resources
       azure_rm_resourcegroup:
         name: "{{ resource_group_name }}"
         state: absent
-        force_delete_nonempty: true
+        force_delete_nonempty: yes
```
Comment (ssbarnea) [inline molecule_azure/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/destroy.yml]: I am sure yamllint would complain about not using recommended yaml booleans, please use them, even if ansible team does not want to conform to newer YAML standard.

Comment (mksha) [reply molecule_azure/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/destroy.yml]: I checked yamllint on destroy.yml , its not failing , but i think we should use `true`. I will make that changes

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**